src/main/java/core/basesyntax/strategy/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.strategy.operation;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+
+public class PurchaseOperation implements OperationHandler {
+    private StorageDao storageDao = new StorageDaoImpl();
```

Comment:
Instantiating StorageDao inside the PurchaseOperation class makes it tightly coupled with the specific implementation of StorageDao. Consider using dependency injection to pass the StorageDao instance to the PurchaseOperation class, which will make your code more modular and easier to test.